pmac/src/lib.rs

```diff
@@ -65,8 +65,9 @@ pub use double::Doublable;
 type Block<N> = GenericArray<u8, N>;
 type ParBlocks<N, M> = GenericArray<GenericArray<u8, N>, M>;
 
-// in future it can be parameter of Pmac
-const LC_SIZE: usize = 16;
+/// Will use only cached table up to 16*2^20 = 16 MB of input data.
```

Comment:
You could calculate these values dynamically for larger sizes. They only get triggered for extremely long strings of zeroes, which happen pretty infrequently.

That said, I'm personally fine with a 16MB cap here.

Reply:
I though checking if value was calculated earlier will add unnecessary branching. Plus we still have to reserve array for values. Hopefully one day we will be able to make it a customizable parameter.

Also note that I do dynamic calculation (see method `get_l`).

Reply:
It would have variable timing, but based on the length of the message and not on any secrets